content/docs/tasks/traffic-management/secure-ingress/sds/index.md

```diff
@@ -0,0 +1,600 @@
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: httpbin
+      labels:
+        app: httpbin
+    spec:
+      ports:
+      - name: http
+        port: 8000
+      selector:
+        app: httpbin
+    ---
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      name: httpbin
+    spec:
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: httpbin
+            version: v1
+        spec:
+          containers:
+          - image: docker.io/citizenstig/httpbin
+            imagePullPolicy: IfNotPresent
+            name: httpbin
+            ports:
+            - containerPort: 8000
+    EOF
+    {{< /text >}}
+
+1.  Create a secret for ingress gateway
+
+    {{< text bash >}}
+    $ kubectl create -n istio-system secret generic httpbin-credential \
+    --from-file=key=httpbin.example.com/3_application/private/httpbin.example.com.key.pem \
+    --from-file=cert=httpbin.example.com/3_application/certs/httpbin.example.com.cert.pem
+    {{< /text >}}
+
+1.  Define a Gateway with a server section for port 443, and specify
+`credentialName` to be `httpbin-credential`, which should be the same as the
+secret name. TLS mode should be specified as SIMPLE.
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: mygateway
+    spec:
+      selector:
+        istio: ingressgateway # use istio default ingress gateway
+      servers:
+      - port:
+          number: 443
+          name: https
+          protocol: HTTPS
+        tls:
+          mode: SIMPLE
+          credentialName: "httpbin-credential" # must be the same as secret
+        hosts:
+        - "httpbin.example.com"
+    EOF
+    {{< /text >}}
+
+1.  Configure routes for traffic entering via the Gateway. Define the same
+`VirtualService`.
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: httpbin
+    spec:
+      hosts:
+      - "httpbin.example.com"
+      gateways:
+      - mygateway
+      http:
+      - match:
+        - uri:
+            prefix: /status
+        - uri:
+            prefix: /delay
+        route:
+        - destination:
+            port:
+              number: 8000
+            host: httpbin
+    EOF
+    {{< /text >}}
+
+1.  Access the `httpbin` service with HTTPS by sending an https request
+
+    {{< text bash >}}
+    $ curl -v -HHost:httpbin.example.com \
+    --resolve httpbin.example.com:$SECURE_INGRESS_PORT:$INGRESS_HOST \
+    --cacert httpbin.example.com/2_intermediate/certs/ca-chain.cert.pem \
+    https://httpbin.example.com:$SECURE_INGRESS_PORT/status/418
+    {{< /text >}}
+
+    The `httpbin` service will return the
+    [418 I'm a Teapot](https://tools.ietf.org/html/rfc7168#section-2.3.3) code.
+
+1.  Replace the credentials for the ingress gateway. You can change the
```

Comment:
Delete the gateway's secret and create a new one to change the ingress gateway's credentials.

Reply:
Done.